packages/deployer/src/deployment.js

```diff
@@ -330,6 +331,11 @@ class Deployment {
           eventArgs.estimateError = err;
         }
 
+        /**
+         * Required as this sets the wallet for each migration
+         */
+        tezos && contract.interfaceAdapter.setWallet(self);
```

Comment:
I think this needs to be

```suggestion
        tezos && (await contract.interfaceAdapter.setWallet(self));
```

@gnidan your thoughts?

Reply:
Looks right to me

Reply:
Applied.